### ISSUE TITLE
Github get additional oauth permissions during login

### ISSUE THREAD

**Issue by geekgonecrazy** [RocketChat/Rocket.Chat.Ops#24]
Currently linking github only gives limited access.  So we will need to get additional permissions approved.

https://help.github.com/articles/connecting-with-third-party-applications/#requesting-updated-permissions

**Comment by Sing-Li** [comment]
I see.  So we probably need {"note":"githubot","scopes":["repo"]}

**Comment by leefaus** [comment]
You can do this through an application key.  The downside is that user will be the one the bot will be acting on as.  So just like in Rocket.Chat, you would probably want to create a Hubot user.

**Comment by Sing-Li** [comment]
I think the app key approach makes the most sense.  @engelgabriel , can you please get that key (or beef up rocketchat's) for us?  
We are somewhat unique that our bot logic can contain client side portion that acts independently - acts as user without app key.  But that is both a high wire act and a hacker-come-get-me invite at this time. Not a wise approach.

**Comment by geekgonecrazy** [comment]
So to be safe here we're going to need to build up some kind of permissions before we implement this then?  Because we for sure do not want just anyone able to perform some operations.

**Comment by engelgabriel** [comment]
I agree with @geekgonecrazy 

So should we do this through an application key for now? So it is restricted to our repos. And filter only users with admin:true until we add the roles layer?

**Comment by Sing-Li** [comment]
Yes.  let's run with it!   And do it 'proper' a little later.  The code base does not look large.

BTW ... public read access does not require  permissions at all - like https://github.com/RocketChat/Rocket.Chat.Ops/issues/25
